scanner/engine/master.cpp

```
@@ -521,10 +522,25 @@ class MasterImpl final : public proto::Master::Service {
     return grpc::Status::OK;
   }
 
+  grpc::Status Shutdown(grpc::ServerContext *context,
+                        const proto::Empty *empty,
+                        Result *result) {
+    trigger_shutdown_ = true;
+    for (auto& w : workers_) {
+      grpc::ClientContext ctx;
+      proto::Empty empty;
+      proto::Result result;
+      w->Shutdown(&ctx, empty, &result);
+    }
+    result->set_success(true);
```

Although currently the worker shutdown never fails, shouldn't we check the result instead of assuming it will always return true?